subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/detection/DefaultTestExecuter.java

```diff
@@ -70,7 +68,7 @@ public DefaultTestExecuter(
     ) {
         this.workerFactory = workerFactory;
         this.actorFactory = actorFactory;
-        this.moduleRegistry = moduleRegistry;
+        this.testClasspathFactory = new TestClasspathFactory(moduleRegistry);
```

Comment:
❓ Should this be created as a service and injected into the constructor rather than instantiated here?

Reply:
I don't think so. Thats making everything way too complicated. The intention was not to have some injectable pluggable system for generating test classpaths. It was so that this logic could be unit tested separately.